src/monstergenerator.cpp

```diff
@@ -770,6 +778,8 @@ void mtype::load( const JsonObject &jo, const std::string &src )
     assign( jo, "armor_pure", armor_pure, strict, 0 );
     assign( jo, "armor_biological", armor_biological, strict, 0 );
 
+    optional( jo, was_loaded, "trap_avoids", trap_avoids, string_id_reader<trap> {} );
```

Comment:
You probably don't need this reader (needed for int ids) but I don't know.

Reply:
This is from where the Json is read and the trap_str_ids are fetched

Reply:
The ids are needed to compare against the trap being triggered to check whether it gets triggered

Reply:
I'm saying this could probably just work as
```suggestion
    optional( jo, was_loaded, "trap_avoids", trap_avoids );
```
I'm fairly sure I've used it that way before in several other places.

But I can't test, and it doesn't hurt to have it here.

Reply:
You are right that is not needed and have made that change